### PR TITLE
Certificate listing api for searching in UI

### PR DESCRIPTION
**Pull request by sarathmohan4**

### CODE REVIEW

**Review by alexrajan3 — APPROVED** [review]

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 1]
```diff
@@ -11140,4 +11140,143 @@ private void getCertificateNamesFromPolicies(String certificatePrefix, List<Stri
...
+
+    /**
+     * To get all certificate names which the user has read/deny/owner permission
+     * @param token
+     * @param userDetails
+     * @param searchText
+     * @return
+     */
+    public ResponseEntity<String> getFullCertificateList(String token, UserDetails userDetails, String searchText) {
+        if (!StringUtils.isEmpty(searchText) && searchText.length() < 3) {
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:11152]: What is the need for this?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 2]
```diff
@@ -11140,4 +11140,143 @@ private void getCertificateNamesFromPolicies(String certificatePrefix, List<Stri
...
+            log.info(JSONUtil.getJSON(ImmutableMap.<String, String>builder()
+                    .put(LogMessage.USER, ThreadLocalContext.getCurrentMap().get(LogMessage.USER))
+                    .put(LogMessage.ACTION, "getFullCertificateList")
+                    .put(LogMessage.MESSAGE, "Trying to get all certificates for normal user from policies")
+                    .put(LogMessage.APIURL, ThreadLocalContext.getCurrentMap().get(LogMessage.APIURL)).build()));
+            String[] policies = policyUtils.getCurrentPolicies(userDetails.getSelfSupportToken(), userDetails.getUsername(), userDetails);
+            certificateList = extractCertificateNameFromPoilcies(policies);
+        }
+        // Filter based on searchText if exists
+        if (!StringUtils.isEmpty(searchText) && searchText.length() >= 3) {
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:11184]: what is the need to filter?

File: tvaultapi/src/main/resources/swaggernotes.properties [hunk 3]
```diff
@@ -1331,4 +1331,7 @@ BuildDetailsController.getBuildDetails.value=To get the build details
 BuildDetailsController.getBuildDetails.notes=<b>This API will fetch the application build details like build version and date of deployment for that particular environment.</b>
```
Comment (sivanag1974) [inline tvaultapi/src/main/resources/swaggernotes.properties:1331]: why have we named the files as swaggernotes.properties?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java (hunk 1 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:11152]: Added for future use

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java (hunk 2 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:11184]: Added for future use

**Pull request merged**